### PR TITLE
`renderTo` overwrites if the container's current `TemplateInstance` has a different `Template`.

### LINKED ISSUE #6: Static text content of node does not change when re-rendered
Consider the following contrived usage of lit:

```javascript
const container = document.createElement('div');

html`<div>foo</div>`.renderTo(container);
html`<div>bar</div>`.renderTo(container);
```

## What I expected

```html
<div>
  <div>bar</div>
</div>
```

## What I got

```html
<div>
  <div>foo</div> <!-- The text content didn't change! -->
</div>
```

### PR DESCRIPTION
**Pull request by bicknellr**
Fixes #6.

### CODE REVIEW

File: src/labs/lit-extended.ts [hunk 1]
```diff
@@ -44,16 +44,22 @@ import { TemplateResult, AttributePart, TemplateInstance, TemplatePart, Part, Te
...
 export function renderExtendedTo(result: TemplateResult, container: Element|DocumentFragment) {
-  let instance = container.__templateInstance as TemplateInstance;
-  if (instance === undefined) {
-    instance = new ExtendedTemplateInstance(result.template);
-    container.__templateInstance = instance;
-    const fragment = instance._clone();
-    instance.update(result.values);
-    container.appendChild(fragment);
-  } else {
+  let instance = container.__templateInstance as ExtendedTemplateInstance|TemplateInstance|undefined;
```
Comment (bicknellr) [inline src/labs/lit-extended.ts]: This doesn't need `ExtendedTemplateInstance|`.

**Comment by bicknellr** [comment]
This doesn't work for `ExtendedTemplateInstance`: if there was an old `TemplateInstance` attached to the container and you attempt to `renderExtendedTo` it, then it won't update the instance.

**Comment by justinfagnani** [comment]
I'm confused by your last comment, are you saying this PR doesn't work in that case, or the code didn't before?

File: src/labs/lit-extended.ts [hunk 2]
```diff
@@ -44,16 +44,22 @@ import { TemplateResult, AttributePart, TemplateInstance, TemplatePart, Part, Te
...
+  if (instance && instance.template === result.template) {
```
Comment (justinfagnani) [inline src/labs/lit-extended.ts]: I think you want to add an instanceof check for `ExtendedTemplateInstance`

File: src/lit-html.ts [hunk 3]
```diff
@@ -48,16 +48,22 @@ export class TemplateResult {
...
-    let instance = container.__templateInstance as TemplateInstance;
-    if (instance === undefined) {
-      instance = new TemplateInstance(this.template);
-      container.__templateInstance = instance;
-      const fragment = instance._clone();
-      instance.update(this.values);
-      container.appendChild(fragment);
-    } else {
+    let instance = container.__templateInstance as TemplateInstance|undefined;
+    if (instance && instance.template === this.template) {
```
Comment (justinfagnani) [inline src/lit-html.ts]: also instanceof check here

**Comment by bicknellr** [comment]
This PR doesn't work in that case. I'm about to push a fix though.

File: src/labs/lit-extended.ts (hunk 2 above)
Comment (justinfagnani) [inline src/labs/lit-extended.ts]: also, explicitly check again `undefined`, it's faster because it skips a ToBoolean on `instance`

Comment (bicknellr) [inline src/labs/lit-extended.ts]: updated

Comment (bicknellr) [inline src/labs/lit-extended.ts]: updated

File: src/lit-html.ts (hunk 3 above)
Comment (bicknellr) [inline src/lit-html.ts]: updated

**Comment by bicknellr** [comment]
`renderExtendedTo` should now cause a new `ExtendedTemplateInstance` to be created, even if a plain `TemplateInstance` is attached to the container and has the same template.

**Pull request merged**